### PR TITLE
Rust-Actix

### PR DESCRIPTION
**Pull request by ingalls**
### Context

Rocket has served us very well over the past couple years and I'm a little sad to move away from it's awesome ergonomics. 

However in the past year actix has come ahead as a clear contender in the space and I think we should migrate as soon as possible
- Remove dependency on rust nightly (huge)
- Move to futures based connections (more concurrency)
- Move to tokio-postgres to avoid blocking workers

cc/ @mapbox/search

### CODE REVIEW

File: src/err/mod.rs [hunk 1]
```diff
@@ -21,18 +21,24 @@ impl HecateError {
...
-    pub fn from_json(code: u16, json: serde_json::Value, safe_error: String, full_error: Option<String>) -> Self {
-        let full_error = match full_error {
-            Some(err) => err,
-            None => safe_error.clone()
-        };
+    pub fn set_json(mut self, json: serde_json::Value) -> Self {
+        self.custom_json = Some(json);
+        self
+    }
+
```
Comment (miccolis) [inline src/err/mod.rs:28]: Extra empty line here.

**Review by miccolis — COMMENTED** [review]
@ingalls I gave this a read through. Saw one very small things and have a question, but at this point I'm pretty unfamiliar with the codebase, as well as both frameworks at play here.

File: src/user/token.rs [hunk 2]
```diff
@@ -0,0 +1,86 @@
+use crate::err::HecateError;
+
+#[derive(Deserialize, Serialize, PartialEq, Debug, Clone)]
```
[the remark this answers is not in the document]
Comment (ingalls) [reply src/user/token.rs:3]: @CyanRook generally speaking only one copy should exist, I added the clone derivation for debugging purposes. I can remove now but the effect on the final binary is negligible.

File: src/bounds/mod.rs [hunk 3]
```diff
@@ -3,7 +3,12 @@ use crate::stream::PGStream;
 
 pub fn set(conn: &impl postgres::GenericConnection, name: &String, feat: &serde_json::Value) -> Result<bool, HecateError> {
     match conn.execute("
-        INSERT INTO bounds (name, geom) VALUES ($1 , ST_Multi(ST_SetSRID(ST_GeomFromGeoJSON($2::JSON->>'geometry'), 4326)))
+        INSERT INTO bounds (name, geom, props)
+            VALUES (
+                $1,
+                ST_Multi(ST_SetSRID(ST_GeomFromGeoJSON($2::JSON->>'geometry'), 4326)),
+                COALESCE(($2::JSON->>'properties')::JSON, '{}'::JSON)::JSONB
```
[the remark this answers is not in the document]
Comment (ingalls) [reply src/bounds/mod.rs:10]: Bug unrelated to actix. A full feature can be passed to the bounds api, and we previously didn't store the meta data into the column that exists to hold it. Not something we use right now but it should work.

**Pull request merged**